### PR TITLE
Adding support for the MQX RTOS.

### PR DESCRIPTION
**Pull request by ocielliottc**
- Use ACE_OS::send and ACE_OS::recv on MQX, instead of write and read.
- Added static casts where pid_t is assumed to be an int.
- Use configurable macro for the "use_select" parameter.
- MQX does not follow conventional standards when it comes to HTONS and NTOHS and must be configured accordingly.
- Modified ACE_Handle_Set so that the use of fd_array is not conditional upon ACE_WIN32.  It is now enabled through ACE_HANDLE_SET_USES_FD_ARRAY.
- Implement UNIX File IO Functions for MQX in MQX_Filesystem.*
- Use the existing implementation to simulate condition variables.
- Use emulation for diropen on MQX as is done for Windows.
- Convert from MQX error to standard error number.
- Use _mqx_exit in ACE_OS::_exit for MQX.
- Added a nop in ACE_OS::ioctl for MQX.
- Added a definition of ACE_TMAIN for MQX so that MQX is initialized and RTCS and the MQX Filesystem code is initialized before calling ace_main_i().
- Fixed a compile error in SOCK_Dgram.cpp introduced months ago.
- Use struct initialization to avoid using the l_linger member directly.
- Modified ACE_Select_Reactor_Base to use a hash map for handle storage based on ACE_SELECT_REACTOR_BASE_USES_HASH_MAP instead of ACE_WIN32.
- Added the IAR compiler to define ACE_NOTREACHED(a) as empty.
- Made ACE_IPPROTO_TCP conditionally defined.
- Added error values required for ACE.
- Skip the wide-character strtok test if ACE_LACKS_WCSTOK is defined in ACE/tests/OS_Test.cpp.
- Added the ability to compile tests so that they output to stderr instead of a file.

### CODE REVIEW

File: ACE/ace/OS_NS_stdio.h [hunk 1]
```diff
@@ -126,6 +126,9 @@ inline ACE_HANDLE ace_fileno_helper (FILE *fp)
 # if defined (fileno)
   return (ACE_HANDLE)fileno (fp);
 # undef fileno
+# elif defined (ACE_LACKS_FILENO)
+  ACE_UNUSED_ARG (fp);
+  return ACE_INVALID_HANDLE;
```
Comment (jwillemsen) [inline ACE/ace/OS_NS_stdio.h]: ACE_NOTSUP_RETURN here?

**Comment by ocielliottc** [comment]
On Thu, Jun 6, 2019 at 11:56 AM Johnny Willemsen <notifications@github.com>
wrote:

> *@jwillemsen* commented on this pull request.
> ------------------------------
>
> In ACE/ace/OS_NS_stdio.h
> <https://github.com/DOCGroup/ACE_TAO/pull/917#discussion_r291278676>:
>
> > @@ -126,6 +126,9 @@ inline ACE_HANDLE ace_fileno_helper (FILE *fp)
>  # if defined (fileno)
>    return (ACE_HANDLE)fileno (fp);
>  # undef fileno
> +# elif defined (ACE_LACKS_FILENO)
> +  ACE_UNUSED_ARG (fp);
> +  return ACE_INVALID_HANDLE;
>
> ACE_NOTSUP_RETURN here?
>

Definitely.

**Comment by jwillemsen** [comment]
This PR added 2 new warnings, see http://buildlogs.remedy.nl/sles150_ace_i686_debug/

```
[Details] /home/build/jenkins/workspace/sles150_ace_i686_debug/BUILD/DOC_ROOT/ACE/ace/SOCK_SEQPACK_Association.cpp: In member function ‘int ACE_SOCK_SEQPACK_Association::abort()’:
[Details] /home/build/jenkins/workspace/sles150_ace_i686_debug/BUILD/DOC_ROOT/ACE/ace/SOCK_SEQPACK_Association.cpp:52:24: warning: missing initializer for member ‘linger::l_linger’ [-Wmissing-field-initializers]
```

**Comment by ocielliottc** [comment]
These are spurious warnings.  Should I add code just to quiet this compiler?

**Comment by jwillemsen** [comment]
Please make a new PR to get rid of the warning again, very likely more platforms will trigger this warning. Hopefully at some point linux will be warning free again so that we can enable warning as error in some CI builds.

**Pull request merged**